decidim-core/lib/decidim/searchable.rb

```diff
@@ -54,6 +54,15 @@ def by_organization(org_id)
         end
       end
 
+      # I am not sure if this needs to be so complicated, or is enough to do a simple:
+      # searchable.searchable_resources.destroy_all
+      after_touch do |searchable|
```

Comment:
Can you add some specs of this? I'd said this should be in `decidim-core/spec/services/decidim/searchable_user_resource_spec.rb`, but if you have any other place more adequate, feel free to add it.
We could try the "`searchable.searchable_resources.destroy_all`" theory, as the `after_destroy` has already a spec there, and we should change it in both places.